api/descriptions/settings/registry_put.md

```diff
@@ -1,32 +1,27 @@
-Specifies the registries to scan.
-The list of registries is set in a single shot.
-Any previous settings are completely overwritten.
+Updates the registries to scan.
```

Comment:
@KICKdesigns The `/settings/registry`, `/settings/vms`, `/settings/coderepos`, and /settings/serverless-scan` are designed to function the same way: you add your account info (the place where your resources live, e.g. an AWS account), we auto-discover the resources in those accounts, then we pass those resources to the scanner for evaluation.

So for this endpoint, the critical fields are:

* `version`
* `registry`
* `credentialID`

The other fields (e.g. repository field, exclusions, etc) are optional, but let you refine the scope of what we auto-discover.

This endpoint works hand-in-hand with the `/policies` endpoints. For the resources that we auto-discover based on `/settings/registries` (where a resource is a container image in this case), the scanner uses the corresponding `/policies` endpoints (in this case `/policies/vulnerability/images` and `/policies/compliance/images`) to assess the resource.

The reason your current curl example works is because an empty `registry` string implicitly refers to Docker Hub (let's document this) and you're pulling from a public repo. The `library` namespace specifies a Docker official image (let's also document this, you can read more about Docker official images here (https://docs.docker.com/docker-hub/official_images/) and the current list is here (https://github.com/docker-library/official-images/tree/master/library).

A more realistic customer setup is a private registry. In this case the customer will need to:

1. Create the appropriate credential in the credentials store.
2. Create the registry setting, referencing the credential.

Let's expand the example for this endpoint to show how this works end-to-end. (create the credentials with the `/credentials` endpoint, get the credential id from the `/credentials` endpoint, and create the registry setting with the "min required" fields (i.e. `version`, `registry`, and `credentialID`).

I'll send you an email with access to one of our private repos for testing.